tests/jax/integration/parameter_server_test.py

```diff
@@ -1,89 +1,357 @@
+        assert jnp.size(network["w"]) != 0
+        assert jnp.size(network["b"]) != 0
+
+    param_without_net = parameter_server.store.parameters.copy()
+    del param_without_net["networks-network_agent"]
+    assert param_without_net == {
+        "trainer_steps": jnp.zeros(1, dtype=jnp.int32),
+        "trainer_walltime": jnp.zeros(1, dtype=jnp.float32),
+        "evaluator_steps": jnp.zeros(1, dtype=jnp.int32),
+        "evaluator_episodes": jnp.zeros(1, dtype=jnp.int32),
+        "executor_episodes": jnp.zeros(1, dtype=jnp.int32),
+        "executor_steps": jnp.zeros(1, dtype=jnp.int32),
+    }
+
+    ############################get_parameters test#####################################
+
+    parameter_server.get_parameters("trainer_steps")
+    assert parameter_server.store.get_parameters == jnp.zeros(1, dtype=jnp.int32)
+
+    parameter_server.get_parameters("networks-network_agent")
+    assert (
+        parameter_server.store.get_parameters
+        == parameter_server.store.parameters["networks-network_agent"]
+    )
 
-    step_var = parameter_server.get_parameters("trainer_steps")
-    assert type(step_var) == np.ndarray
-    assert step_var[0] == 0
+    # get multiple params
+    parameter_server.get_parameters(["executor_episodes", "executor_steps"])
+    assert parameter_server.store.get_parameters == {
+        "executor_episodes": jnp.zeros(1, dtype=jnp.int32),
+        "executor_steps": jnp.zeros(1, dtype=jnp.int32),
+    }
 
-    parameter_server.set_parameters({"trainer_steps": np.ones(1, dtype=np.int32)})
-    assert parameter_server.get_parameters("trainer_steps")[0] == 1
+    ############################set_parameters test#####################################
+    params = {
+        "trainer_steps": jnp.zeros(2, dtype=jnp.int32),
+        "trainer_walltime": jnp.zeros(2, dtype=jnp.float32),
+        "evaluator_steps": jnp.zeros(2, dtype=jnp.int32),
+        "evaluator_episodes": jnp.zeros(2, dtype=jnp.int32),
+        "executor_episodes": jnp.zeros(2, dtype=jnp.int32),
+        "executor_steps": jnp.zeros(2, dtype=jnp.int32),
+    }
 
-    parameter_server.add_to_parameters({"trainer_steps": np.ones(1, dtype=np.int32)})
-    assert parameter_server.get_parameters("trainer_steps")[0] == 2
+    parameter_server.set_parameters(params)
+    list_param = [
+        "trainer_steps",
+        "trainer_walltime",
+        "evaluator_steps",
+        "evaluator_episodes",
+        "executor_episodes",
+        "executor_steps",
+    ]
+    for param in list_param:
+        assert jnp.array_equal(
+            parameter_server.store.parameters[param], jnp.array([0, 0])
+        )
+
+    # set non existing param
+    param1: Dict[str, str] = {"wrong_param": "test"}
+    with pytest.raises(AssertionError):
+        assert parameter_server.set_parameters(param1)
+
+    ############################add_to_parameters_test#####################################
+    param2: Dict[str, Any] = {
+        "trainer_steps": jnp.array([1, 3]),
```

Comment:
Issue here is that manually updating the params messes with the terminators (which we may not end up using though)